app/components/navheader/index.js

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Box, Flex } from 'theme-ui';
+import _ from 'lodash';
+
+import Name from './Name';
+import DemographicInfo from './DemographicInfo';
+import PatientMenuOptions from './MenuOptions/Patient';
+import ClinicianMenuOptions from './MenuOptions/Clinician';
+
+const UploadLaunchOverlay = require('../../components/uploadlaunchoverlay');
+
+const innerContainerStyleProps = {
+  px: 4, 
+  py: 3, 
+  sx: {
+    columnGap: 5,
+    flexWrap: 'wrap',
+    justifyContent: ['center', 'space-between'],
+    alignItems: 'center',
+    rowGap: 2,
+  }
+};
+
+const NavHeader = ({ patient, isUserPatient, trackMetric, permsOfLoggedInUser }) => {
+  const history = useHistory();
+  const [isUploadOverlayOpen, setIsUploadOverlayOpen] = useState(false);
+
+  if (!patient?.profile) return null; // not available immediately on component mount
+
+  const { permissions } = patient;
+  
+  const canUpload = _.isEmpty(permissions) === false && permissions.root || _.has(permsOfLoggedInUser, 'upload');
+  const canShare = _.isEmpty(permissions) === false && permissions.root;
```

Comment:
I'm not sure of all of the possible states of the permissions, but I basically copied these conditions verbatim from `app/components/navbarpatientcard/navbarpatientcard.js` so hoping that everything lines up.